### DISCUSSION TITLE
GHG Attribution in calculateSectorPurchasedbySectorSourcedImpact function & Confirmation of Output

### DISCUSSION THREAD

**Discussion by ranastasiadis** [USEPA/useeior#267] [Q&A] [answered]
Hello, 

#### Context
I originally had a similar [question to #67 in the Supply Chain repo](https://github.com/USEPA/supply-chain-factors/discussions/67) but I believe it is better posed here. I followed the linked code in 67 to the flowsa repo, but I believe my question to be more around the underlying model values in this repository. I have used the same NAICS codes as the original 67 post in this. I have also reviewed the [sources and sinks document starting at 4.24](https://www.epa.gov/system/files/documents/2023-02/US-GHG-Inventory-2023-Main-Text.pdf). Please redirect me if this is still the incorrect location for this question.

#### Process
Using the calculateSectorPurchasedbySectorSourcedImpact function I have generated an entire sector x sector (to my understanding) matrix to look up the differences between the listed industries who had "HFC emissions [...] significantly higher than other related industries". Below are the NAICS that matched (Couldn't find a clean one for the 517100 listed NAICS)

![image](https://github.com/USEPA/useeior/assets/139541116/21d427ee-9ca2-4ecf-9131-75e7cc8db2cc)

### Code and File
``` 
library(useeior)
seeAvailableModels()

model <- buildModel("USEEIOv2.1-422")


sector2sector_impact <- calculateSectorPurchasedbySectorSourcedImpact(y = model$DemandVectors$vectors$`2012_US_Consumption_Complete`,
                                                                      model,
                                                                      indicator = "Greenhouse Gases")
                                                                      
write.csv(sector2sector_impact, "SECTOR_X_SECTOR_IMPACTS_BY_GHG.csv", row.names = TRUE, fileEncoding = "UTF-8")
```

[SECTOR_X_SECTOR_IMPACTS_BY_GHG.csv](https://github.com/USEPA/useeior/files/13649759/SECTOR_X_SECTOR_IMPACTS_BY_GHG.csv)

### Y/N Questions:

    1. Has the above R code generated a complete and accurate Sector x Sector crosswalk for the specified model configuration?

    2 (If yes to 1): Can absolute units of demand be compared across varying purchasing sectors to understand where something like the larger HFC contribution from `333415` may be?
    3 (if no to 1): Can this sort of model insight be gotten to in the USEEIOR repo?
    4 (if no to 1 and 3): Is flowsa this source? And if so, is it required we locate and ingest the refrigeration industry data source itself to compare across industries? 

    5. Is comparing GHGs across sectors this way in absolutes sound or is there a normalization factor I may be missing?

Thank you and enjoy your year end, 

Ryan

**Comment by bl-young** [comment]
> model <- buildModel("USEEIOv2.1-422")

Note this model was erroneously included in the package and has not yet been completed. I would not recommend using it.

**Reply by ranastasiadis** [reply]
Thank you for the FYI on this, I will adjust my code!

**Comment by bl-young** [comment, marked as answer]
In the `SectorPurchasedbySectorSourcedImpact` matrix, the rows represent the commodity purchased (in your case representing the complete consumption demand vector), while the columns reflect the sector where the emissions occur.

> 1. Has the above R code generated a complete and accurate Sector x Sector crosswalk for the specified model configuration?

Yes

> 2 (If yes to 1): Can absolute units of demand be compared across varying purchasing sectors to understand where something like the larger HFC contribution from `333415` may be?

I'm not sure I understand this question - the screenshot above can be interpreted as: "The total consumption of 515200 results in global warming emissions of 69,948,273 in the 333415" which is answering a different question than was in [post 67](https://github.com/USEPA/supply-chain-factors/discussions/67). But yes, the results do scale with the quantity of demand. Note that currently you can not create a sector by sector matrix for a specific flow, but the [function](https://github.com/USEPA/useeior/blob/master/R/CalculationFunctions.R#L262-L269) could easily be tweaked to do so

> 3 (if no to 1): Can this sort of model insight be gotten to in the USEEIOR repo?
> 4 (if no to 1 and 3): Is flowsa this source? And if so, is it required we locate and ingest the refrigeration industry data source itself to compare across industries? 

Perhaps I'm misunderstanding the question. The flowsa datasets will reveal the direct emissions from any sector. If you want to understand the supply chain emissions resulting from a purchase you'll need to use USEEIO matrices.

> 5. Is comparing GHGs across sectors this way in absolutes sound or is there a normalization factor I may be missing?

Can you reframe your specific question? The question in the referenced post was related to _direct_ emissions of HFCs from some media sectors. And the explanation was that to estimate emissions resulting from air conditioning, we use data for purchases of air conditioning equipment as a proxy for attributing those emissions.

The matrix you produced here accounts for overall quantity of demand. If you simply want to just compare emissions factors per dollar you can use the `M` or `N` matrices.

**Reply by ranastasiadis** [reply]
Certainly, thank you for the detailed response. I had originally attempted to list out my questions in yes/no form but may have abstracted my point out entirely. 

#### Base Question

"When looking to understand contribution of GHG emissions per dollar spend for any given sector, what is the best way/repo/function to dig into the inter-sector purchases that contributed to this?"

#### Example Answer Logic

"It is believed that GHG emissions are higher in general in the model for `515200` vs those in `(515100, 517110, 512100)`.  Is demand in any one supplier-industry (with a higher emissions footprint) driving a portion of this?

HFCs are an ODS that largely comes from Fugitive Emissions, something that comes from a limited set of sources in large quantities, making an easily identifiable test case in `333415` as, for HFCs, refrigeration is one of said sources.

Thus, using the matrix attached, one industry driving emissions in another industry would be seen if emissions from purchases of 515200 (row) are 'significantly' higher than the others in the `333415`  sector (col)"

_____

The hope being that then, this could be used to understand and validate YoY trends being seen vs changes in the dollar factor over year that comes from the SEF repo. But via understanding where inter-industry purchases driving emissions are occuring.

>Note that currently you can not create a sector by sector matrix for a specific flow, but the [function](https://github.com/USEPA/useeior/blob/master/R/CalculationFunctions.R#L262-L269) could easily be tweaked to do so

Could you elaborate on an example of what a 'specific flow' would be? Is this a data source from the upstream FLOWSA repo as was loaded by the model?

>"The total consumption of 515200 results in global warming emissions of 69,948,273 in the 333415"

Is this in metric tonnes of CO2e?

Appreciate the help!

Ryan

**Reply by bl-young** [reply]
> Is this in metric tonnes of CO2e?

You can confirm in `model$indicators` but I believe this is kg CO2e

**Reply by bl-young** [reply]
> "When looking to understand contribution of GHG emissions per dollar spend for any given sector, what is the best way/repo/function to dig into the inter-sector purchases that contributed to this?"

The datasets that come out of flowsa (and the underlying models) will help you understand the makeup of the _direct_ emissions from a sector. Things like, how much of the emissions are process emissions vs combustion emissions, what fuels are responsible for the combustion emissions, etc.

Anything related to the supply chain or emissions driven by purchases are explained by details only available in useeio.

**Reply by bl-young** [reply]
> "It is believed that GHG emissions are higher in general in the model for `515200` vs those in `(515100, 517110, 512100)`. Is demand in any one supplier-industry (with a higher emissions footprint) driving a portion of this?

For a question like this, you might look to a few places. The [B matrix](https://github.com/USEPA/useeior/blob/master/format_specs/Model.md#b) in a USEEIO model would show you the direct emissions _per dollar_. You may also want to look to the [Totals by Sector (TbS)](https://github.com/USEPA/useeior/blob/master/format_specs/Model.md#totals_by_sector) itself which reflects the emissions directly out of flowsa[^1] before they are normalized to sector output. In both of these cases, these reflect direct emissions from the sector excluding the supply chain.

However, depending on the sector, the upstream/supply chain emissions may be significant. The [M matrix](https://github.com/USEPA/useeior/blob/master/format_specs/Model.md#M) reflects the total (upstream + direct) emissions from each sector. If the upstream emissions are substantial, you would then need to do further analysis to identify which sectors (purchases) are driving those upstream emissions.

[^1]: This is a simplification. The data that comes out of flowsa is slightly more resolved (by NAICS) while once it is in useeio it has been condensed to the BEA schema. The flowsa datasets will also have a bit more metadata to help understand the underlying sources of the emissions.

**Reply by bl-young** [reply]
> Could you elaborate on an example of what a 'specific flow' would be? Is this a data source from the upstream FLOWSA repo as was loaded by the model?

Specific flow meaning `Carbon dioxide` or `HFC-125`, instead of your current output which reflects characterized impacts for `Global warming`
